_test/template.response.test.php

```diff
@@ -0,0 +1,85 @@
+<?php
+//======================================================================
+// TEMPLATE RESPONSE TESTS
+//======================================================================
+
+require_once(__DIR__ . '/../DokuwikiXMLExport.php');
+require_once(__DIR__ . '/../admin.php');
+require_once(__DIR__ . '/../_test/Helper.php');
+
+class template_response_test extends DokuWikiTest
```

Comment:
Test 0, 1, multiple, a lot.

Reply:
Test Umlaute, special characters etc. do not mess up your output.

Reply:
> Test Umlaute, special characters etc. do not mess up your output.

DokuWiki does not allow pages with umlauts or special characters. All special characters are set to `_` and all umlauts are set to their corresponding form e.g. `ä = ae`.

Reply:
Keep logic as much as possible out of the template. This will make it easier to test and make the responsibilities of the different components much clearer.